### PR TITLE
doc: gs: add sections to updating tools

### PR DESCRIPTION
**Pull request by greg-fer**
Expanded Updating tools and repositories with TCM-related sections.
NCSDK-14128.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/gs_updating.rst
Comment (wiba-nordic) [inline doc/nrf/gs_updating.rst:286]:
```suggestion
.. |method_note| replace:: Use the method corresponding to the way you installed the |NCS|, as described in the following sections.
```

Comment (Avalei) [inline doc/nrf/gs_updating.rst:286]:
```suggestion
.. |method_note| replace:: Use the method corresponding to the way you installed the |NCS|, as described in the following sections.
```

**Review by Avalei — CHANGES_REQUESTED** [review]
Is @divipillai removing the SES section as a separate task or should that removal be included in this PR?

File: doc/nrf/gs_updating.rst
Comment (divipillai) [reply doc/nrf/gs_updating.rst:54]: Yes, removing the SES section will be part of PR https://github.com/nrfconnect/sdk-nrf/pull/7484 and for image - We have a Jira task created - https://projecttools.nordicsemi.no/jira/browse/NCSDK-15030.

**Comment by greg-fer** [comment]
@Avalei , @wiba-nordic , @divipillai , changes applied.

**Review by wiba-nordic — APPROVED** [review]

**Review by Avalei — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: doc/nrf/gs_updating.rst
Comment (trond-snekvik) [inline doc/nrf/gs_updating.rst:39]: Hey woah, this isn't quite right. VSC provides no mechanism for upgrading the west tool itself. The section in the link at the bottom is correct though. VSC lets you switch your NCS version and its dependencies through the source control panel, but the tool itself must be updated through other means

Comment (Avalei) [reply doc/nrf/gs_updating.rst:39]: Are the "other means" of updating west basically just `pip3 install -U west` and equivalents for each platform?

Comment (greg-fer) [reply doc/nrf/gs_updating.rst:39]: I think that section relates more to updating everything that west manages, not west itself. I guess I misunderstood that at first.

Comment (greg-fer) [reply doc/nrf/gs_updating.rst:39]: @trond-snekvik , pushed an edit, please check again.

**Comment by greg-fer** [comment]
@trond-snekvik , could you check this task? I'm waiting for your approval to do the final squash.

**Review by trond-snekvik — APPROVED** [review]
LGTM! Sorry for the delay.

**Pull request merged**
